block.go

```diff
@@ -151,6 +151,9 @@ type BlockMeta struct {
 
 	// Information on compactions the block was created from.
 	Compaction BlockMetaCompaction `json:"compaction"`
+
+	// Version of the index format
+	Version int `json:"version"`
```

Comment:
Generally 👍  for making this explicit here. But I'm wondering whether we actually need to bump the format. The serialisation is certainly different with the padding, but it seems like `DecoderV2` below doesn't do anything differently than `DecoderV1`.

Reply:
Yup, fair enough. I'll just rename it to `Decoder` instead.

Reply:
However, the main issue with having two different formats/versions was how to make them configurable. Currently, the value is being hardcoded as `1`. However, picking it up from `BlockMeta` itself makes the test fail as there is no value specified in the test files and the default value is `0`. How can that be fixed?